docs/cdktf/typescript/data-sources/http.md

```diff
@@ -0,0 +1,242 @@
+ * See https://cdk.tf/provider-generation for more details.
+ */
+import { DataHttp } from "./.gen/providers/http/";
+class MyConvertedCode extends TerraformStack {
+  constructor(scope: Construct, name: string) {
+    super(scope, name);
+    new DataHttp(this, "example", {
+      request_headers: [
+        {
+          Accept: "application/json",
+        },
+      ],
+      url: "https://checkpoint-api.hashicorp.com/v1/check/terraform",
+    });
+    new DataHttp(this, "example_head", {
+      method: "HEAD",
+      url: "https://checkpoint-api.hashicorp.com/v1/check/terraform",
+    });
+    new DataHttp(this, "example_post", {
+      method: "POST",
+      request_body: "request body",
+      url: "https://checkpoint-api.hashicorp.com/v1/check/terraform",
+    });
+  }
+}
+
+```
+
+## Usage with Postcondition
+
+[Precondition and Postcondition](https://www.terraform.io/language/expressions/custom-conditions)
+checks are available with Terraform v1.2.0 and later.
+
+```typescript
+// Code generated by 'cdktf convert' - Please report bugs at https://cdk.tf/bug
+import { Construct } from "constructs";
+import { TerraformSelf, Fn, TerraformStack } from "cdktf";
+/*
+ * Provider bindings are generated by running `cdktf get`.
+ * See https://cdk.tf/provider-generation for more details.
+ */
+import { DataHttp } from "./.gen/providers/http/";
+class MyConvertedCode extends TerraformStack {
+  constructor(scope: Construct, name: string) {
+    super(scope, name);
+    new DataHttp(this, "example", {
+      lifecycle: {
+        postcondition: [
+          {
+            condition: Fn.contains(
+              [201, 204],
+              TerraformSelf.getAny("status_code")
+            ),
+            errorMessage: "Status code invalid",
+          },
+        ],
+      },
+      request_headers: [
+        {
+          Accept: "application/json",
+        },
+      ],
+      url: "https://checkpoint-api.hashicorp.com/v1/check/terraform",
+    });
+  }
+}
+
+```
+
+## Usage with Precondition
+
+[Precondition and Postcondition](https://www.terraform.io/language/expressions/custom-conditions)
+checks are available with Terraform v1.2.0 and later.
+
+```typescript
+// Code generated by 'cdktf convert' - Please report bugs at https://cdk.tf/bug
+import { Construct } from "constructs";
+import { Fn, TerraformStack } from "cdktf";
+/*
+ * Provider bindings are generated by running `cdktf get`.
+ * See https://cdk.tf/provider-generation for more details.
+ */
+import { DataHttp } from "./.gen/providers/http/";
+import { Uuid } from "./.gen/providers/random/uuid";
+class MyConvertedCode extends TerraformStack {
+  constructor(scope: Construct, name: string) {
+    super(scope, name);
+    /*The following providers are missing schema information and might need manual adjustments to synthesize correctly: random.
+    For a more precise conversion please use the --provider flag in convert.*/
+    const dataHttpExample = new DataHttp(this, "example", {
```

Comment:
Similar question about the `missing schema` info message for `random`: https://github.com/hashicorp/terraform-provider-time/pull/216#discussion_r1261662860